sonarqube-10.yaml

```diff
@@ -12,6 +12,9 @@ package:
       - bash
       - bash-binsh # some operations depend on /bin/sh
       - sonarqube-startup
+      # java native library used on startup
+      - systemd-dev
+      - zstd-dev
 
```

Comment:
@joshrwolf  ouch, does it really need -dev packages because it does dlopen by .so$ name instead of something better? I would have hoped, we can avoid shipping headers at runtime.

Reply:
I think so? here is the stack trace, lmk if there's another way

```
2024.10.18 16:34:07 WARN  es[][o.e.n.NativeAccess] Unable to load native provider. Native methods will be disabled.
java.lang.UnsatisfiedLinkError: Unable to load library 'zstd':
libzstd.so: cannot open shared object file: No such file or directory
libzstd.so: cannot open shared object file: No such file or directory
Native library (linux-aarch64/libzstd.so) not found in resource path ()
        at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:307) ~[jna-5.12.1.jar:?]
        at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:467) ~[jna-5.12.1.jar:?]
        at com.sun.jna.Native.register(Native.java:1774) ~[jna-5.12.1.jar:?]
        at org.elasticsearch.nativeaccess.jna.JnaZstdLibrary.<init>(JnaZstdLibrary.java:32) ~[?:?]
        at org.elasticsearch.nativeaccess.lib.NativeLibraryProvider.getLibrary(NativeLibraryProvider.java:58) ~[elasticsearch-native-8.15.2.jar:?]
        at org.elasticsearch.nativeaccess.AbstractNativeAccess.<init>(AbstractNativeAccess.java:29) ~[elasticsearch-native-8.15.2.jar:?]
        at org.elasticsearch.nativeaccess.PosixNativeAccess.<init>(PosixNativeAccess.java:28) ~[elasticsearch-native-8.15.2.jar:?]
        at org.elasticsearch.nativeaccess.LinuxNativeAccess.<init>(LinuxNativeAccess.java:19) ~[elasticsearch-native-8.15.2.jar:?]
        at org.elasticsearch.nativeaccess.NativeAccessHolder.<clinit>(NativeAccessHolder.java:28) ~[elasticsearch-native-8.15.2.jar:?]
        at org.elasticsearch.nativeaccess.NativeAccess.instance(NativeAccess.java:22) ~[elasticsearch-native-8.15.2.jar:?]
        at org.elasticsearch.bootstrap.Elasticsearch.initializeNatives(Elasticsearch.java:285) ~[elasticsearch-8.15.2.jar:?]
        at org.elasticsearch.bootstrap.Elasticsearch.initPhase2(Elasticsearch.java:170) ~[elasticsearch-8.15.2.jar:?]
        at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:75) ~[elasticsearch-8.15.2.jar:?]
        Suppressed: java.lang.UnsatisfiedLinkError: libzstd.so: cannot open shared object file: No such file or directory
                at com.sun.jna.Native.open(Native Method) ~[jna-5.12.1.jar:?]
                at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:197) ~[jna-5.12.1.jar:?]
                at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:467) ~[jna-5.12.1.jar:?]
                at com.sun.jna.Native.register(Native.java:1774) ~[jna-5.12.1.jar:?]
                at org.elasticsearch.nativeaccess.jna.JnaZstdLibrary.<init>(JnaZstdLibrary.java:32) ~[?:?]
                at org.elasticsearch.nativeaccess.lib.NativeLibraryProvider.getLibrary(NativeLibraryProvider.java:58) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.nativeaccess.AbstractNativeAccess.<init>(AbstractNativeAccess.java:29) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.nativeaccess.PosixNativeAccess.<init>(PosixNativeAccess.java:28) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.nativeaccess.LinuxNativeAccess.<init>(LinuxNativeAccess.java:19) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.nativeaccess.NativeAccessHolder.<clinit>(NativeAccessHolder.java:28) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.nativeaccess.NativeAccess.instance(NativeAccess.java:22) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.bootstrap.Elasticsearch.initializeNatives(Elasticsearch.java:285) ~[elasticsearch-8.15.2.jar:?]
                at org.elasticsearch.bootstrap.Elasticsearch.initPhase2(Elasticsearch.java:170) ~[elasticsearch-8.15.2.jar:?]
                at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:75) ~[elasticsearch-8.15.2.jar:?]
        Suppressed: java.lang.UnsatisfiedLinkError: libzstd.so: cannot open shared object file: No such file or directory
                at com.sun.jna.Native.open(Native Method) ~[jna-5.12.1.jar:?]
                at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:210) ~[jna-5.12.1.jar:?]
                at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:467) ~[jna-5.12.1.jar:?]
                at com.sun.jna.Native.register(Native.java:1774) ~[jna-5.12.1.jar:?]
                at org.elasticsearch.nativeaccess.jna.JnaZstdLibrary.<init>(JnaZstdLibrary.java:32) ~[?:?]
                at org.elasticsearch.nativeaccess.lib.NativeLibraryProvider.getLibrary(NativeLibraryProvider.java:58) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.nativeaccess.AbstractNativeAccess.<init>(AbstractNativeAccess.java:29) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.nativeaccess.PosixNativeAccess.<init>(PosixNativeAccess.java:28) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.nativeaccess.LinuxNativeAccess.<init>(LinuxNativeAccess.java:19) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.nativeaccess.NativeAccessHolder.<clinit>(NativeAccessHolder.java:28) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.nativeaccess.NativeAccess.instance(NativeAccess.java:22) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.bootstrap.Elasticsearch.initializeNatives(Elasticsearch.java:285) ~[elasticsearch-8.15.2.jar:?]
                at org.elasticsearch.bootstrap.Elasticsearch.initPhase2(Elasticsearch.java:170) ~[elasticsearch-8.15.2.jar:?]
                at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:75) ~[elasticsearch-8.15.2.jar:?]
        Suppressed: java.io.IOException: Native library (linux-aarch64/libzstd.so) not found in resource path ()
                at com.sun.jna.Native.extractFromResourcePath(Native.java:1145) ~[jna-5.12.1.jar:?]
                at com.sun.jna.NativeLibrary.loadLibrary(NativeLibrary.java:281) ~[jna-5.12.1.jar:?]
                at com.sun.jna.NativeLibrary.getInstance(NativeLibrary.java:467) ~[jna-5.12.1.jar:?]
                at com.sun.jna.Native.register(Native.java:1774) ~[jna-5.12.1.jar:?]
                at org.elasticsearch.nativeaccess.jna.JnaZstdLibrary.<init>(JnaZstdLibrary.java:32) ~[?:?]
                at org.elasticsearch.nativeaccess.lib.NativeLibraryProvider.getLibrary(NativeLibraryProvider.java:58) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.nativeaccess.AbstractNativeAccess.<init>(AbstractNativeAccess.java:29) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.nativeaccess.PosixNativeAccess.<init>(PosixNativeAccess.java:28) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.nativeaccess.LinuxNativeAccess.<init>(LinuxNativeAccess.java:19) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.nativeaccess.NativeAccessHolder.<clinit>(NativeAccessHolder.java:28) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.nativeaccess.NativeAccess.instance(NativeAccess.java:22) ~[elasticsearch-native-8.15.2.jar:?]
                at org.elasticsearch.bootstrap.Elasticsearch.initializeNatives(Elasticsearch.java:285) ~[elasticsearch-8.15.2.jar:?]
                at org.elasticsearch.bootstrap.Elasticsearch.initPhase2(Elasticsearch.java:170) ~[elasticsearch-8.15.2.jar:?]
                at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:75) ~[elasticsearch-8.15.2.jar:?]
2024.10.18 16:34:07 WARN  es[][o.e.n.NativeAccess] Cannot check if running as root because native access is not available
```